task/idle/src/main.rs

```diff
@@ -31,9 +28,12 @@ fn main() -> ! {
             // which could wake some higher-priority task.
             cortex_m::asm::wfi();
         }
+
         // RISC-V has wfi, but unfortunately it is an illegal instruction if
-        // called from User mode, so instead we spin on a timer call.
-        #[cfg(target_arch = "riscv32")]
-        while sys_get_timer().now > 0 {}
+        // called from User mode.
+        #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+        unsafe {
+            riscv::asm::nop();
```

Comment:
Disassembly of the current rv64 version of this suggested we were generating a jump directly to itself. I added this to be truer to the intent of the above, so now we generate a nop, j, rather than just a j. Critically, this avoids a context switch per iteration, in favour of the nop.

Reply:
Fair enough, I think making the disassembly easier to read is good reason!